[React] Extended notifications

imobachgs opened the pull request:
This PR is basically an evolution of #319. I have dropped cogo-toast (nice project, btw) and implemented a small *duck* to handle notifications and adding a new component to display them. Moreover, I have added timeout support.

Pending stuff (separate PRs)

- [x] Good looking notifications :wink:
  
  ![new-notifications png](https://user-images.githubusercontent.com/1691872/82667815-76941180-9c30-11ea-9a62-52a1083f2f1c.png)

- [x] Allow the user to block a given type of notifications ("do not show again")

![do-not-show-again.png](https://user-images.githubusercontent.com/15836/83254002-e57be800-a1a5-11ea-9931-e7868182aadb.png)

dgdavid commented on src/components/Notifications/Notification.js:
```diff
@@ -21,9 +21,12 @@ function Notification({ notification }) {
+        <button className="text-blue-600 font-bold mr-2" onClick={() => dispatch(actions.close(notification.id))}>
           Dismiss
         </button>
+        <button className="text-blue-600 font-bold mr-2" onClick={() => dispatch(actions.blacklistAndClose(notification))}>
```
For now, I'd use `text-blue-300` for this action.

https://tailwindcss.com/docs/customizing-colors#default-color-palette

ancorgs commented on src/state/ducks/notifications.js:
```diff
@@ -10,23 +10,112 @@
  *
  * @todo Remember which notifications should be 'silenced'.
```
This is not longer `@todo` isn't it?

ancorgs commented on src/components/Notifications/Notification.js:
```diff
@@ -0,0 +1,37 @@
+          Dismiss
+        </button>
+
+        {notification.type &&
```
I guess we can refine this in the future, not sure whether having type or not is the right criteria for allowing the "do not show again".

imobachgs replied:
Yes, you are right. I came to the same conclusion. But now that I've added support for notification actions, this code is not relevant anymore :)

imobachgs commented:
I am not happy about the fact of having to pass the notification ID to the `dispatchAction` action creator. I added such an ID to the action itself, but then I changed my mind. By now, I think it is good enough. We can refine later.

dgdavid commented on src/state/ducks/notifications.test.js:
```diff
@@ -68,26 +74,14 @@ describe('action creators', () => {
+    it('adds the message type to the black list', () => {
       const store = mockStore({ notifications: initialState });
-      store.dispatch(actionCreators.notifyMessage('You have been muted.'));
+      console.log("ACTIOH", actionCreators.blacklist('muted'));
```
A debugging leftover?

imobachgs replied:
:flushed:
